kythe/go/extractors/cmd/rust/rust_project_to_kzip.go

```diff
@@ -0,0 +1,552 @@
+/*
+ * Copyright 2025 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"kythe.io/kythe/go/platform/kzip"
+	"kythe.io/kythe/go/platform/vfs"
+	"kythe.io/kythe/go/util/vnameutil"
+	apb "kythe.io/kythe/proto/analysis_go_proto"
+	spb "kythe.io/kythe/proto/storage_go_proto"
+)
+
+// the central state for the rust-project extractor
+type extractor struct {
+	project             rustProject
+	projectRoot         string
+	corpus              string
+	rules               vnameutil.Rules
+	kzipWriter          kzipWriterInterface
+	requiredInputsCache *requiredInputsCache
+}
+
+// kzipWriterInterface defines the methods needed from a kzip writer.
+type kzipWriterInterface interface {
+	AddUnit(cu *apb.CompilationUnit, index *apb.IndexedCompilation_Index) (string, error)
+	AddFile(r io.Reader) (string, error)
+	Close() error // Add if Close is needed by the caller of writeCrate, though not writeCrate itself
+}
+
+// source is the set of directories which may have source files for this crate,
+// as well as directories which are explicitly excluded from containing necessary source files.
+type source struct {
+	IncludeDirs []string `json:"include_dirs"`
+	ExcludeDirs []string `json:"exclude_dirs"`
+}
+
+// dep is a dependency of a Crate
+type dep struct {
+	CrateId crateId `json:"crate"`
+	Name    string  `json:"name"`
+}
+
+type crate struct {
+	RootModule        string   `json:"root_module"`
+	Edition           string   `json:"edition"`
+	Deps              []dep    `json:"deps"`
+	Cfg               []string `json:"cfg"`
+	CompilerArgs      []string `json:"compiler_args"`
```

Comment:
I don't think this attribute exists, per https://rust-analyzer.github.io/book/non_cargo_based_projects.html

Reply:
I suspect r-a may reject files with unknown fields, which might explain the lack of xrefs you're seeing.

The indexer results should have some clues if this is the case.